packages/address-book-controller/src/AddressBookController.ts

```diff
@@ -120,8 +129,15 @@ export class AddressBookController extends BaseController<
    * @param chainId - Chain id identifies the current chain.
    * @param memo - User's note about address.
    * @returns Boolean indicating if the address was successfully set.
+   * @param addressType - Contact's address type.
    */
-  set(address: string, name: string, chainId = '1', memo = '') {
+  set(
+    address: string,
+    name: string,
+    chainId = '1',
+    memo = '',
+    addressType = AddressType.unknown,
```

Comment:
We should cover this new parameter in a unit test. I see that existing unit tests already show that this works correctly when `addressType` is not set, but we don't have any tests that set this to a value other than unknown.

Reply:
I added new unit tests for the nonAccounts and contractAccounts address types.
Unknown and externallyOwnedAccounts address types were covered before.

Reply:
Thanks!